src/settings/global_user.rs

```diff
@@ -17,7 +26,94 @@ pub enum GlobalUser {
 
 impl GlobalUser {
     pub fn new() -> Result<Self, failure::Error> {
-        get_global_config()
+        let environment = Environment::with_whitelist(ENV_VAR_WHITELIST.to_vec());
+
+        let config_path = default_config_file().expect("could not find global config directory");
+
+        GlobalUser::build(environment, config_path)
+    }
+
+    fn build<T: 'static + QueryEnvironment>(
+        environment: T,
+        config_path: PathBuf,
+    ) -> Result<Self, failure::Error>
+    where
+        T: config::Source + Send + Sync,
+    {
+        if let Some(user) = Self::from_env(environment) {
+            user
+        } else {
+            Self::from_file(config_path)
+        }
+    }
+
+    fn from_env<T: 'static + QueryEnvironment>(
+        environment: T,
+    ) -> Option<Result<Self, failure::Error>>
+    where
+        T: config::Source + Send + Sync,
+    {
+        // if there's some problem with gathering the environment,
+        // or if there are no relevant environment variables set,
+        // fall back to config file.
+        if environment.empty().unwrap_or(true) {
+            None
+        } else {
+            let mut s = config::Config::new();
+            s.merge(environment).ok();
+
+            Some(GlobalUser::from_config(s))
+        }
+    }
+
+    fn from_file(config_path: PathBuf) -> Result<Self, failure::Error> {
+        let mut s = config::Config::new();
+
+        let config_str = config_path
+            .to_str()
+            .expect("global config path should be a string");
+
+        // Skip reading global config if non existent
+        // because envs might be provided
+        if config_path.exists() {
+            log::info!(
+                "Config path exists. Reading from config file, {}",
+                config_str
+            );
+            s.merge(config::File::with_name(config_str))?;
+        } else {
```

Comment:
Should we just print a warning and not bail here? If we bail here, it seems that the fallback case (allowing people to just use env vars without config files, as they would in CI) would fail.